react/week2/my_first_app/src/index.js

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+//import App from './App';
+import * as serviceWorker from './serviceWorker';
+
+const root = document.getElementById('root');
+
+const todosArray = [
+  {
+    "id": 1,
+    "description": "Get out of bed"
+  },
+  {
+    "id": 2,
+    "description": "Brush teeth"
+  },
+  {
+    "id": 3,
+    "description": "Eat breakfast"
+  }, 
+  {
+    "id": 4,
+    "description": "Understand State and props"
+  },
+  {
+    "id": 5,
+    "description": "Passing the state through props"
+  }
+];
+class Header extends React.Component {
+  render() {
+    return <h1>Todolist</h1>
+  }
+}
+//I cannot pass a props that is in a state in one component and
+//set that props again in state in different component
+class TodoList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isChecked: false,
+      checked: []
+    };
+  }
+
+  handleCheckboxChange = (event) => {
+    const temp = event.target.checked
+    //Eventhough the item.id is int when it is passed inside name attribute in input tag
+    // it changes to string thats why parseInt
+    const specificTodo = parseInt(event.target.name)
+    //When I check the box my temp becomes true and executes if part
+    if (temp) {
+      //When checked I push that particular id to my checked array and update the state
+      this.state.checked.push(specificTodo)
+      this.setState({
+        isChecked: temp,
+        checked: this.state.checked
+      })
+    } else {
+      //When I uncheck, I get the index of the unchecked todo and remove it from checked array and update the state
+      const index = this.state.checked.indexOf(specificTodo)
+      this.state.checked.splice(index, 1);
+      this.setState({
+        isChecked: temp,
+        checked: this.state.checked
+      })
+    }
+  }
+
+  render() {
+    const array = this.props.todos;
+    const result = array.map((item) => {
+      return (
+        <div className='todo-list-display' key={item.id}>
```

Comment:
You should have another component for the todo items (e.g `TodoItem` or `Todo`) and use the props and callbacks accordingly to change its state.

Reply:
Implemented